### ISSUE TITLE
[Bug]: mpl with py3.10.1 - Interactive figures - Constrain pan/zoom to x/y axis not work

### ISSUE THREAD

**Issue by edge-python** [matplotlib/matplotlib#22028]
### Bug summary

mpl with py3.10.1 - Interactive figures - Constrain pan/zoom to x/y axis not work

### Code for reproduction

```python
Simple Plot

https://matplotlib.org/stable/gallery/lines_bars_and_markers/simple_plot.html
```

### Actual outcome

Interactive figures - Constrain pan/zoom to x/y axis not work

Despite the key being pressed, both axes change

### Expected outcome

i expect the functionality:

Navigation keyboard shortcuts

https://matplotlib.org/stable/users/explain/interactive.html

Constrain pan/zoom to x axis  -  hold x when panning/zooming with mouse
Constrain pan/zoom to y axis  -  hold y when panning/zooming with mouse

### Additional information

I've tested a few versions:

py 3.8 + mpl 3.4.1 and 3.5.1  o.k.
py 3.9 + mpl 3.4.1 and 3.5.1  o.k.

but:

py 3.10.1 + mpl 3.4.1 and 3.5.1 --> Constrain pan/zoom  only to x or y axis NOT WORK !

What is the reason?

Thank you

### Operating system

Win 10

### Matplotlib Version

 different 3.4.1 ... 3.5.1

### Matplotlib Backend

TkAgg

### Python version

Python 3.10.1

### Installation

pip

**Comment by jklymak** [comment]
Sounds like a Tk problem   Are you able to test on Qt5?

**Comment by edge-python** [comment]
I am inexperienced with Qt5
I tested with the following code 

```
# --------------------------------------------------------------------------------
# https://www.pythonguis.com/tutorials/plotting-matplotlib/
import sys
import matplotlib
matplotlib.use('Qt5Agg')

from PyQt5 import QtCore, QtGui, QtWidgets

from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg, NavigationToolbar2QT as NavigationToolbar
from matplotlib.figure import Figure


class MplCanvas(FigureCanvasQTAgg):

    def __init__(self, parent=None, width=5, height=4, dpi=100):
        fig = Figure(figsize=(width, height), dpi=dpi)
        self.axes = fig.add_subplot(111)
        super(MplCanvas, self).__init__(fig)


class MainWindow(QtWidgets.QMainWindow):

    def __init__(self, *args, **kwargs):
        super(MainWindow, self).__init__(*args, **kwargs)

        sc = MplCanvas(self, width=5, height=4, dpi=100)
        sc.axes.plot([0,1,2,3,4], [10,1,20,3,40])

        # Create toolbar, passing canvas as first parament, parent (self, the MainWindow) as second.
        toolbar = NavigationToolbar(sc, self)

        layout = QtWidgets.QVBoxLayout()
        layout.addWidget(toolbar)
        layout.addWidget(sc)

        # Create a placeholder widget to hold our toolbar and canvas.
        widget = QtWidgets.QWidget()
        widget.setLayout(layout)
        self.setCentralWidget(widget)

        self.show()


app = QtWidgets.QApplication(sys.argv)
w = MainWindow()
app.exec_()
# --------------------------------------------------------------------------------
```


**Result**

py 3.10.1 + mpl 3.5.1  --> Constrain pan/zoom only to x or y axis **NOT WORK !**
py 3.9 + mpl 3.4.1  -----> Constrain pan/zoom only to x or y axis **NOT WORK !**

**???**

**Comment by edge-python** [comment]
there is another problem:

**the buttons are difficult to activate**
(applies to all such as Home/Reset, Back, Forward, Pan/Zoom, etc.)

not the whole area reacts, only the extreme left side of the button 

--> **this problem only with  TkAgg**, not with Qt5

**Comment by jklymak** [comment]
You should have just been able to do 

```python
import matplotlib
matplotlib.use('Qt5Agg')
import matplotlib.pyplot as plt

plt.plot(range(10))
plt.show()
```

and work from there.  If you are embedding Matplotlb in your own GUI, then that is a different set of debugging steps...

**Comment by edge-python** [comment]
ok, with your Qt5 example **it works as it should** 
(no idea why not with my previous Qt5 example )

but what is the problem with the first example with the TkAgg backend?

**Comment by edge-python** [comment]
I did further research (with the first example with the TkAgg backend):

I uninstalled py3.10.1 and installed py3.10.0   
py3.10.0 + mpl 3.4.1 or 3.5.1 **o.K.,  it works as it should** and **the buttons are easy to activate**

then I installed the update py3.10.1 
py3.10.1 + mpl 3.4.1 or 3.5.1 --> Constrain pan/zoom only to x or y axis **NOT WORK !** and **the buttons are difficult to activate**

**Comment by tacaswell** [comment]
This sounds like there are upstream issues with the tk bindings (as they ship with Python itself).

**Comment by ebardie** [comment]
As mentioned on #22255, #22077 does "fix" #22255.

I'm not sure that as it stands #22077 is taking the right approach for addressing the problem though:

1. the problem is probably caused by changes lower in the stack, so ideally (I know, I know) would be resolved there too.
2. #22077 changes behaviour, i.e. the user needs to click to set the focus. Ideally, the focus should still be there from the start. 
3. #22255 occurs on Windows but not Linux (I've no Mac to test on), and #22077 affects all platforms.

At the least, don't remove the `focus_set()` call from `__init__()` for unaffected platforms (i.e intead wrap with `if sys.platform != "win32":`), and wrap the `focus_set()` in `button_press_event()` with `if sys.platform == "win32":`.

**Comment by tacaswell** [comment]
Closed by #22077

**Comment by ctcob** [comment]
I'm having the exact same problem in Windows 10 (up to date), PyQt5.15.2, and Python 3.9.10.
